### ISSUE TITLE
time: Parse error messages may be confusing/misleading

### ISSUE THREAD

**Issue by extemporalgenome** [golang/go#56730]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.19.3 darwin/amd64
</pre>

### Does this issue reproduce with the latest release?

Yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=""
GOARCH="amd64"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
</pre></details>

### What did you do?

https://go.dev/play/p/PUywlZ7-BE5

```go
func main() {
	_, err := time.Parse("2006-01-02", "22-10-25")
	fmt.Println(err)
}
```

### What did you expect to see?

```
parsing time "22-10-25" as "2006-01-02": cannot parse "22" as "2006"
```

### What did you see instead?

```
parsing time "22-10-25" as "2006-01-02": cannot parse "0-25" as "2006"
```

It appears that this the parser is doing some blocking or chunking that gets misaligned in the error reporting when fewer than the expected number of digits appear in the input.

**Comment by bcmills** [comment]
Does this still reproduce at HEAD? @dsnet has landed a bunch of `Parse` fixes during the 1.20 development cycle.

**Comment by extemporalgenome** [comment]
Unfortunately it's still an issue:

```
> ./bin/go version
go version devel go1.20-cb5534c1c6 Wed Nov 16 02:02:26 2022 +0000 linux/amd64
> ./bin/go run /tmp/timebug.go
parsing time "22-10-25" as "2006-01-02": cannot parse "0-25" as "2006"
```
